ws-client/src/client.rs

```diff
@@ -550,6 +588,26 @@ async fn background_task(
 					stop_subscription(&mut sender, &mut manager, unsub).await;
 				}
 			}
+
+			// User called `register_notification` on the front-end.
+			Either::Left((Some(FrontToBack::RegisterNotification(reg)), _)) => {
+				log::trace!("[backend] registering notification handler: {:?}", reg.method);
+				let (subscribe_tx, subscribe_rx) = mpsc::channel(max_notifs_per_subscription);
+
+				if manager.insert_notification_handler(&reg.method, subscribe_tx).is_ok() {
+					reg.send_back
+						.send(Ok((subscribe_rx, reg.method)))
+						.expect("error sending response for notification handler");
+				} else {
+					let _ = reg.send_back.send(Err(Error::MethodAlreadyRegistered(reg.method)));
+				}
+			}
+
+			// User called `on_notification` on the front-end.
```

Comment:
Is this comment correct? Why does calling `on_notification` cause de-registration?